### ISSUE TITLE
Please support defaulting to default profile of root when IAM role fails

### ISSUE THREAD

**Issue by eric-tucker** [adamsb6/s3_file#91]
We're running s3_file on a non EC2 instance that has /root/.aws/config and /root/.aws/credentials filled in with a default profile to simulate having an IAM role best we can.

s3_file doesn't gracefully handle this and just dies.  It would be nice if it handled this in a manner similar to how the CLI does where we can just call without any profile or credentials values like this:

```
aws s3 cp ...
```

**Comment by joekiller** [comment]
Yeah supporting the IAM chain would be helpful but without the AWS SDK it is a bit of a pain to support. I dunno how it goes with the "lightweight part" if we download the s3 sdk and include it. but we could just take the core and the s3 files.

Consider this a good thread to throw in an opinion.
